awx/ui_next/src/screens/Credential/CredentialDetail/CredentialDetail.jsx

```diff
@@ -48,19 +60,33 @@ function CredentialDetail({ i18n, credential }) {
       setContentError(null);
       setHasContentLoading(true);
       try {
-        const {
-          data: { inputs: credentialTypeInputs, managed_by_tower },
-        } = await CredentialTypesAPI.readDetail(credential_type.id);
+        const [
+          {
+            data: { inputs: credentialTypeInputs, managed_by_tower },
+          },
+          {
+            data: { results: loadedInputSources },
+          },
+        ] = await Promise.all([
+          CredentialTypesAPI.readDetail(credential_type.id),
+          CredentialsAPI.readInputSources(credentialId, { page_size: 200 }),
```

Comment:
since we keep these functions in our api file now, we could do the recursive to last page thing there...would be a bit more elegant than keeping it in the views like we had to do in the old app.  but i'm also fine for keeping things simple like this.

Reply:
@jlmitch5 see if https://github.com/ansible/awx/pull/7281/commits/834cfb82f8da49003633743599640d405c92cf47 is what you're looking for.  I like this pattern for requests that we know will be made recursively.